### PR TITLE
[SHIPA-1020] Update Circle CI Orb with pool to framework

### PR DESCRIPTION
**Pull request by itsksaurabh**
This PR adds support for frameworks, adds integration tests and refactors the structure of the project.  I've fixed a number of bugs and linted the shell scripts and yaml so this is all green now. @murphybytes

### CODE REVIEW

**Review by parkerduckworth — APPROVED** [review]
Very long PR but I managed to get through it 🙂

**Review by c12o16h1 — APPROVED** [review]
LGTM

**Comment by itsksaurabh** [comment]
@murphybytes @c12o16h1 Could you please release a new orb with this change such that it doesn’t break older ones?

**Comment by vivek** [comment]
We are changing pool to framework. Please make sure it merged after shipa 1.2.1 release with pool to framework changes so that we can ensure orb is working with 1.2.0

File: .circleci/config.yml [hunk 1]
```diff
@@ -15,7 +15,7 @@ parameters:
     default: false
   dev-orb-version:
     type: string
-    default: "dev:alpha"
+    default: "1.0.1"
```
Comment (murphybytes) [inline .circleci/config.yml:18]: This doesn't change, it should remain 'dev:alpha'

File: src/commands/app-create.yml [hunk 2]
```diff
@@ -14,7 +14,7 @@ parameters:
     description: Shipa team name
     type: env_var_name
   pool:
-    default: SHIPA_POOL
+    default: SHIPA_FRAMEWORK
     description: Shipa pool name
```
Comment (murphybytes) [inline src/commands/app-create.yml:18]: If you're going to change the environment variable, you need to change the description as well.

**Review by murphybytes — CHANGES_REQUESTED** [review]
I haven't checked if there were any changes in the shipa API itself.  If there were you may need to modify the shipa-ci python app.

File: .circleci/config.yml (hunk 1 above)
Comment (itsksaurabh) [reply .circleci/config.yml:18]: dev:alpha is a trial feature that has already expired. Please see https://shipa-io.slack.com/archives/CS3KL4BK7/p1611837102166700

File: src/commands/app-create.yml (hunk 2 above)
Comment (itsksaurabh) [reply src/commands/app-create.yml]: agree. fixed

File: .circleci/config.yml [hunk 3]
```diff
@@ -18,77 +23,63 @@ parameters:
...
 jobs:
-  # This job is an example of an integration testing job.
-  # This job should execute a command from your orb and verify
-  # the output is as expected, otherwise the job should fail.
-  #
-  # Rename this job and add more as needed.
-  #
-  integration-test-1:
-    executor: orb-tools/ubuntu
+  integration-tests:
```
Comment (c12o16h1) [inline .circleci/config.yml:26]: What about keeping some comments explaining that this is just an example of the job? Without that, it takes some time to understand why we set params explicitly here, and what is it overall...

Comment (c12o16h1) [reply .circleci/config.yml:26]: Or rename `integration-tests` to `example-job` or so

**Review by c12o16h1 — APPROVED** [review]
Not critical, but please pay some attention to my comments

File: .circleci/config.yml (hunk 3 above)
Comment (murphybytes) [reply .circleci/config.yml:26]: Actually, these are tests that run every time you check in

**Pull request merged**